modules/core/shared/src/main/scala-3/io/circe/derivation/package.scala

```diff
@@ -25,26 +25,50 @@ private[circe] inline final def summonLabels[T <: Tuple]: List[String] =
     case _: EmptyTuple => Nil
     case _: (t *: ts)  => constValue[t].asInstanceOf[String] :: summonLabels[ts]
 
+@deprecated("Use summonEncoders(derivingForSum: Boolean) instead", "0.14.7-BL")
 private[circe] inline final def summonEncoders[T <: Tuple](using Configuration): List[Encoder[_]] =
+  summonEncoders(false)
+
+private[circe] inline final def summonEncoders[T <: Tuple](inline derivingForSum: Boolean)(using
+  Configuration
+): List[Encoder[_]] =
   inline erasedValue[T] match
     case _: EmptyTuple => Nil
-    case _: (t *: ts)  => summonEncoder[t] :: summonEncoders[ts]
+    case _: (t *: ts)  => summonEncoder[t](derivingForSum) :: summonEncoders[ts](derivingForSum)
 
+@deprecated("Use summonEncoder(derivingForSum: Boolean) instead", "0.14.7-BL")
 private[circe] inline final def summonEncoder[A](using Configuration): Encoder[A] =
+  summonEncoder(false)
+
+private[circe] inline final def summonEncoder[A](inline derivingForSum: Boolean)(using Configuration): Encoder[A] =
   summonFrom {
     case encodeA: Encoder[A] => encodeA
-    case _: Mirror.Of[A]     => ConfiguredEncoder.derived[A]
+    case m: Mirror.Of[A] =>
+      inline if (derivingForSum) ConfiguredEncoder.derived[A]
+      else error("Failed to find an instance of Encoder[" + typeName[A] + "]")
   }
 
+@deprecated("Use summonDecoders(derivingForSum: Boolean) instead", "0.14.7-BL")
```

Comment:
BL?

Reply:
Whoops sorry, short for my company name where I was considering publishing a separate version of this to use. Updated